Source/WebCore/page/scrolling/ScrollingCoordinatorTypes.cpp

```diff
@@ -43,6 +43,19 @@ void RequestedScrollData::merge(RequestedScrollData&& other)
             other.requestedDataBeforeAnimatedScroll = requestedDataBeforeAnimatedScroll;
             break;
         }
+    } else if (other.requestType == ScrollRequestType::DeltaUpdate && animated == ScrollIsAnimated::No) {
+        switch (requestType) {
+        case ScrollRequestType::PositionUpdate: {
+            other.requestType = ScrollRequestType::PositionUpdate;
+            other.scrollPositionOrDelta = std::get<FloatPoint>(scrollPositionOrDelta) + std::get<FloatSize>(other.scrollPositionOrDelta);
```

Comment:
This line can similarly be

```
std::get<FloatPoint>(other.scrollPositionOrDelta) += std::get<FloatSize>(other.scrollPositionOrDelta);
```

Reply:
I don't think this would work for this case because `other.scrollPositionOrDelta` would have a `FloatSize` since it is a delta update. This is a special case were we have to convert the incoming delta update to a position update when merging these two requests together.

Reply:
Yes, you are exactly right.